Headers/DebugServer2/Support/POSIX/ELFSupport.h

```diff
@@ -32,8 +32,16 @@ class ELFSupport {
                           NotesHeader &nhdr, ByteVector &id);
 
   template <typename ELFHeader, typename SectionHeader>
-  static bool ReadSectionHeader(int fd, const ELFHeader &ehdr, SectionHeader &shdr,
-                                size_t idx);
+  static bool ReadSectionHeader(int fd, const ELFHeader &ehdr,
+                                SectionHeader &shdr, size_t idx);
+
+  template <typename ELFHeader, typename SectionHeader>
+  static bool ReadStringTable(int fd, const ELFHeader &ehdr,
+                              SectionHeader &shdr, std::vector<char> &table);
```

Comment:
I'm not sure how I feel about this. This is effectively a full copy of the string table. Do we _need_ a copy? Is the mapped view not guaranteed to outlive this the reference? That is, more specifically, can we get away with a `std::span<char> &`?